File: NET/Aliens.Sharp/API/ITriggering.cs
using System;

namespace Example.API
{
    public interface ITriggering : IDisposable
    {
        public delegate bool PCallBack(int hWnd, string lParam);

        bool EnumWindows(PCallBack callback, int count);

        public event ThresholdHandler ThresholdReached;

        void StartPub(int count);

        public class ThresholdEventArgs : EventArgs
        {
            public int Threshold { get; set; }
            public DateTime TimeReached { get; set; }
        }

        public delegate void ThresholdHandler(object sender, ThresholdEventArgs e);
    }
}

File: JVM/aliens-java/src/main/java/org/example/api/ITriggering.java
package org.example.api;

public interface ITriggering extends AutoCloseable {

	interface PCallBack {
		boolean invoke(int hWnd, String lParam);
	}

	boolean enumWindows(PCallBack callback, int count);

	void addThresholdReached(ThresholdHandler handler);

	void removeThresholdReached(ThresholdHandler handler);

	interface ThresholdHandler {
		void invoke(Object sender, ThresholdEventArgs e);
	}

	void startPub(int count);
}
